### PR TITLE
[FW-1875] fixed undefined window error

### PR DESCRIPTION
**Pull request by sebaholesz**
| Q             | A
| ------------- | ---
|Description, reason for the PR| The undefined window error was resolved by addressing it in two different ways. First, the gtmSafePushEvent function was wrapped in a try-catch block with exception logging to handle any errors that might occur. Second, toast messages are now configured to log only on the client side, preventing server-side logging issues. Additionally, the code was refactored to replace raw window conditions with canUseDom or isServer checks, making the code more readable and easier to maintain.
|New feature| No
|[BC breaks](https://docs.shopsys.com/en/latest/contributing/backward-compatibility-promise/)| No
|Fixes issues| ... 
|Have you read and signed our [License Agreement for contributions](https://www.shopsys.com/license-agreement)?| Yes

----
:globe_with_meridians: Live Preview:
  - https://sh-undefined-window-fix.odin.shopsys.cloud
  - https://cz.sh-undefined-window-fix.odin.shopsys.cloud

### CODE REVIEW

File: project-base/storefront/helpers/isServer.ts [hunk 1]
```diff
@@ -1,12 +1,7 @@
...
+import { canUseDom } from './canUseDom';
 
-export const isServer = (): boolean =>
-    // eslint-disable-next-line @typescript-eslint/no-unnecessary-condition
-    !(
-        typeof window !== 'undefined' &&
-        typeof window.document === 'object' &&
-        typeof window.document.createElement === 'function'
-    );
+export const isServer = (): boolean => !canUseDom();
```
Comment (tvikito) [inline project-base/storefront/helpers/isServer.ts]: What about rename `canUseDom` to `isClient`. And then we can remove this `isServer` and use it like `!isClient`?

**Review by tvikito — APPROVED** [review]

File: project-base/storefront/helpers/isServer.ts (hunk 1 above)
Comment (sebaholesz) [reply project-base/storefront/helpers/isServer.ts]: Im good with that 😁 done

File: project-base/storefront/helpers/isClient.ts [hunk 2]
```diff
@@ -0,0 +1,6 @@
+export const isClient = (): boolean =>
+    !!(
+        typeof window !== 'undefined' &&
+        typeof window.document === 'object' &&
+        typeof window.document.createElement === 'function'
+    );
```
Comment (tvikito) [inline project-base/storefront/helpers/isClient.ts]: Hmm, actually one more think which I am wondering about. Couldn't this be simple variable? 🤔

```suggestion
export const isClient = !!(
        typeof window !== 'undefined' &&
        typeof window.document === 'object' &&
        typeof window.document.createElement === 'function'
    );
```

File: docs/storefront/gtm/intro.md [hunk 3]
```diff
@@ -153,16 +153,16 @@ const getPageInfoForBrandDetailPage = (
...
 ```ts
-export const gtmSafePushEvent = (
-  event: GtmEventInterface<GtmEventType, unknown>
-): void => {
-  if (canUseDom()) {
-    window.dataLayer = window.dataLayer ?? [];
-    window.dataLayer.push(event);
-  }
+export const gtmSafePushEvent = (event: GtmEventInterface<GtmEventType, unknown>): void => {
+    if (canUseDom()) {
```
Comment (tvikito) [inline docs/storefront/gtm/intro.md]: Just last piece here

```suggestion
    if (isClient()) {
```

**Review by tvikito — APPROVED** [review]

File: docs/storefront/gtm/intro.md (hunk 3 above)
Comment (sebaholesz) [reply docs/storefront/gtm/intro.md]: Thanks, changed

File: project-base/storefront/helpers/isClient.ts (hunk 2 above)
Comment (sebaholesz) [reply project-base/storefront/helpers/isClient.ts]: It can, nice catch.

**Pull request merged**